smac/utils/sklearn_wrapper.py

```diff
@@ -0,0 +1,329 @@
+"""
+author = "Jan N. van Rijn"
+license = "3-clause BSD"
+"""
+
+import numpy as np
+
+from collections import defaultdict
+from functools import partial
+from typing import Union, Callable, Dict, List
+
+from ConfigSpace import Configuration
+from ConfigSpace.hyperparameters import CategoricalHyperparameter, UniformIntegerHyperparameter, UniformFloatHyperparameter
+
+from sklearn.base import is_classifier, clone, BaseEstimator
+from sklearn.model_selection._split import check_cv, BaseCrossValidator
+from sklearn.model_selection._search import BaseSearchCV
+from sklearn.model_selection._validation import _fit_and_score
+from sklearn.utils.fixes import rankdata
+from sklearn.utils.fixes import MaskedArray
+from sklearn.utils.validation import indexable
+from sklearn.metrics.scorer import check_scoring
+
+from smac.configspace import ConfigurationSpace
+from smac.facade.smac_facade import SMAC
+from smac.scenario.scenario import Scenario
+
+
+class ModelBasedOptimization(BaseSearchCV):
+    """
+    Scikit-Learn wrapper for SMAC
+    """
+    def __init__(self, estimator: BaseEstimator, param_distributions: Dict[str, List], n_iter: int = 10,
+                 scoring: Union[str, Callable, None] = None, fit_params: Union[Dict, None] = None, iid: bool = True,
+                 refit: bool = True, cv: Union[BaseCrossValidator, int] = None, verbose: int = 0,
+                 random_state: Union[int, np.random.RandomState] = None, error_score: Union[str, int] = 'raise',
+                 return_train_score: bool = True):
+        """Scikit-learn wrapper for Sequential Model Based Optimization (SMAC). Is useful as
+        this allows to use the various Scikit-learn interfaces, for example to OpenML.org
```

Comment:
I know what you want to say, but this sentence makes little sense to me, maybe `This allows to use SMAC together with various tools relying on the scikit-learn API, such as OpenML.org.`